src/imitation/scripts/analyze.py

```diff
@@ -49,14 +49,15 @@ def _gather_sacred_dicts(
     sacred_dirs = itertools.chain.from_iterable(
         sacred_util.filter_subdirs(source_dir) for source_dir in source_dirs
     )
-    sacred_dicts = []
+    sacred_dicts_list = []
 
     for sacred_dir in sacred_dirs:
         try:
-            sacred_dicts.append(sacred_util.SacredDicts.load_from_dir(sacred_dir))
+            sacred_dicts_list.append(sacred_util.SacredDicts.load_from_dir(sacred_dir))
         except json.JSONDecodeError:
             warnings.warn(f"Invalid JSON file in {sacred_dir}", RuntimeWarning)
 
+    sacred_dicts: Iterable = sacred_dicts_list
```

Comment:
I get the pros and cons of each. It's hard to visually keep track of type changes and it can lead to confusion. A fix could be forcing a new annotation every time you change the type to aid clarity, but this defeats type inference. Unlikely they will change this though, I'm sure there was an extensive debate about this issue at some point ;)